apps/server/src/shared/repo/user/user.repo.integration.spec.ts

```diff
@@ -399,4 +399,45 @@ describe('user repo', () => {
 			expect(user.id).not.toBeNull();
 		});
 	});
+
+	describe('delete', () => {
+		const setup = async () => {
+			const user1: User = userFactory.buildWithId();
+			const user2: User = userFactory.buildWithId();
+			const user3: User = userFactory.buildWithId();
+			await em.persistAndFlush([user1, user2, user3]);
+
+			return {
+				user1,
+				user2,
+				user3,
+			};
+		};
+		it('should delete user', async () => {
+			const { user1, user2, user3 } = await setup();
+			const deleteResult = await repo.deleteUser(user1.id);
+			expect(deleteResult).toEqual(1);
+
+			const result1 = await em.find(User, { id: user1.id });
+			expect(result1).toHaveLength(0);
+
+			const result2 = await repo.findById(user2.id);
+			expect(result2).toMatchObject({
+				firstName: user2.firstName,
+				lastName: user2.lastName,
+				email: user2.email,
+				roles: user2.roles,
+				school: user2.school,
+			});
```

Comment:
I prefer checking the strict equality, as `toMatchObject()` checks if the object contains the subset of specified object fields (and I like to verify that, e.g., no unexpected field has been added). But it's just a remark, I know `toMatchObject()` is also used in other places across the project so it's ok.